File: Data Structures & Algorithms/invert-a-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        def dfs_invert(node):
            if node == None:
                return node
            [node.right, node.left] = [node.left, node.right];
            dfs_invert(node.left)
            dfs_invert(node.right)
        dfs_invert(root)
        return root;
        

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-1.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @param {number} k
     * @return {number}
     */
    kthSmallest(root, k) {
        let counter = 0;
        let result = null;
        function inOrderTraversal(node) {
            if(!node || result !== null) return;
            if(node.left) inOrderTraversal(node.left);
            counter++;
            if(counter === k) {
                result = node.val;
            }
            if(node.right) inOrderTraversal(node.right);
        }
        inOrderTraversal(root)
        return result;
    }
}
